src/table.js

```diff
@@ -355,6 +355,8 @@ function alignof(base = {}, data, columns) {
 }
 
 function type(data, column) {
+  // duck-type Arrow table date fields
+  if (String(data?.schema?.fields?.find?.(d => d?.name === column)).endsWith("<MILLISECOND>")) return "date";
```

Comment:
This would be cleaner pulled out to a separate function that first tests whether `data` is an Arrow table, like we do in the notebook standard library:

https://github.com/observablehq/stdlib/blob/746ca2e69135df6178e4f3a17244def35d8d6b20/src/arrow.js#L4-L16

Also, I think it’d be cleaner/more explicit/robust to check the numeric data codes rather than using string matching. I’ll have to look up how to do that…